protocol-units/settlement/mcr/contracts/src/MCR.sol

```diff
@@ -205,7 +316,7 @@ contract MCR {
             BlockCommitment memory blockCommitment = commitments[blockHeight][validatorAddress];
```

Comment:
This might be an unnecessary optimisation at this point, but consider instead of looping. Another constant-time op, as a nested mapping. Again for gas. Where the `uint256` is the blockheight
`mapping(uint256 => mapping(address => BlockCommitment)) public blockCommitments;`

Reply:
I'm not sure about this one; we'll want to benchmark. 

The reason for the loop here is that we need to find each commitment at each block height. We could store a separate set of `mapping(uint256 => EnumerableBytes32)` and `mapping(bytes32 => BlockCommitment` then iterate over just the unique commitments. But, this ends up bloating storage which is also costly. 

The current operations within the loop become constant time following the change to `getEpochStakeAt`. With the above, we end up moving these constant time operations up into the commitment insertion logic. I think we basically end up paying the same either way.

Reply:
Hmm yea, it will be expensive either way, was more of a curiosity comment.